trin-cli/src/main.rs

```diff
@@ -9,92 +10,94 @@ use std::path::{Path, PathBuf};
+#[derive(Args, Debug, PartialEq)]
+#[command(name = "json-rpc", about = "Run JSON-RPC commands against a trin node")]
 struct JsonRpc {
     /// IPC path of target JSON-RPC endpoint.
-    #[structopt(default_value(DEFAULT_WEB3_IPC_PATH), long)]
+    #[arg(default_value = DEFAULT_WEB3_IPC_PATH, long)]
     ipc: PathBuf,
 
     /// JSON-RPC method (e.g. discv5_routingTableInfo).
-    #[structopt(required = true)]
+    #[arg(required = true)]
     endpoint: String,
 
     /// Comma-separated list of JSON-RPC parameters.
-    #[structopt(long, use_delimiter = true)]
+    #[arg(long, value_delimiter = ',')]
     params: Option<Vec<String>>,
 }
 
 // TODO: Remove clippy allow once a variant with non-"Block" prefix is added.
 /// Encode a content key.
-#[derive(StructOpt)]
+#[derive(Subcommand, Debug, PartialEq)]
 #[allow(clippy::enum_variant_names)]
 enum EncodeKey {
     /// Encode the content key for a block header.
     BlockHeader {
         /// Hex-encoded block hash (omit '0x' prefix).
-        #[structopt(long)]
+        #[arg(long)]
         block_hash: H256,
     },
     /// Encode the content key for a block body.
     BlockBody {
         /// Hex-encoded block hash (omit '0x' prefix).
-        #[structopt(long)]
+        #[arg(long)]
         block_hash: H256,
     },
     /// Encode the content key for a block's transaction receipts.
     BlockReceipts {
         /// Hex-encoded block hash (omit '0x' prefix).
-        #[structopt(long)]
+        #[arg(long)]
         block_hash: H256,
     },
 }
 
-#[derive(StructOpt)]
+#[derive(Args, Debug, PartialEq)]
+#[command(name = "create-dashboard")]
 #[allow(clippy::enum_variant_names)]
 struct DashboardConfig {
-    #[structopt(default_value = "http://localhost:3000")]
+    #[arg(default_value = "http://localhost:3000")]
     grafana_address: String,
 
-    #[structopt(default_value = "admin")]
+    #[arg(default_value = "admin")]
     grafana_username: String,
 
-    #[structopt(default_value = "admin")]
+    #[arg(default_value = "admin")]
     grafana_password: String,
 
-    #[structopt(default_value = "http://host.docker.internal:8545")]
+    #[arg(default_value = "http://host.docker.internal:8545")]
     trin_rpc_address: String,
 
-    #[structopt(default_value = "http://host.docker.internal:9090")]
+    #[arg(default_value = "http://host.docker.internal:9090")]
     prometheus_address: String,
 }
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    match Trin::from_args() {
-        Trin::JsonRpc(rpc) => json_rpc(rpc),
-        Trin::EncodeKey(content_key) => encode_content_key(content_key),
-        Trin::CreateDashboard(dashboard_config) => create_dashboard(dashboard_config),
+fn main() {
+    let args = Trin::parse();
+    match args {
+        Trin::JsonRpc(rpc) => json_rpc(rpc).unwrap(),
+        Trin::EncodeKey(content_key) => encode_content_key(content_key).unwrap(),
+        Trin::CreateDashboard(dashboard_config) => create_dashboard(dashboard_config).unwrap(),
     }
 }
```

Comment:
```suggestion
fn main() -> Result<(), Box<dyn std::error::Error>> {
    match Trin::parse() {
        Trin::JsonRpc(rpc) => json_rpc(rpc),
        Trin::EncodeKey(content_key) => encode_content_key(content_key),
        Trin::CreateDashboard(dashboard_config) => create_dashboard(dashboard_config),
    }
}
```
We should only use unwrap during tests is there any reason you added unwrap and removed the return?

If you want you can just add the Result back and do match Trin::parse() and the rest of the PR looks good